cmd/flagsaver/flagsaver_test.go

```diff
@@ -0,0 +1,67 @@
+package flagsaver
+
+import (
+	"testing"
+	"time"
+
+	"flag"
+)
+
+var (
+	intFlag      = flag.Int("int_flag", 123, "test integer flag")
+	strFlag      = flag.String("str_flag", "foo", "test string flag")
+	durationFlag = flag.Duration("duration_flag", 5*time.Second, "test duration flag")
+)
+
+// TestRestore checks that flags are saved and restore correctly.
+// Checks are performed on flags with both their default values and with explicit values set.
+// Only a subset of the possible flag types are currently tested.
+func TestRestore(t *testing.T) {
+	tests := []struct {
+		// Test name
+		name string
+		// Name of flag to save and restore.
+		flag string
+		// The value the flag should have when saved. If empty, this indicates the flag should have its default value.
+		oldValue string
+		// The value the flag should have just before being restored to oldValue.
+		newValue string
+	}{
+		{"RestoreDefaultIntValue", "int_flag", "", "666"},
```

Comment:
nit: Please use field labels. It's clearer, saves you from specifying defaults, and saves us from having to touch all structs if a new field is added.